File: PJS5_POC_Cpp/Sources/Robot/Furnace.cpp
#include "Furnace.h"

namespace ROBOT
{
	FURNACE::FURNACE(const int iId, const std::string& sName, const int iArmor, const int iLifePoint, const int iRestartLimit)
		: IPARTS(iId, sName, iArmor, iLifePoint),
		_iRestartLimit(iRestartLimit)
	{
	}

	FURNACE::~FURNACE()
	{
	}
}

File: PJS5_CSharp/PJS5_CSharp/Sources/Robot/Furnace.cs
namespace PJS5_CSharp.Sources.Robot
{
        public class FURNACE : IPARTS
        {   
            private readonly int _iRestartLimit = -1;

            public FURNACE(int iId, string sName, int iArmor, int iLifePoint, int iRestartLimit)
                : base(iId, sName, iArmor, iLifePoint)
            {
                _iRestartLimit = iRestartLimit;
            }

            public int GetRestartLimit()
            {
                return _iRestartLimit;
            }
        }
    }

